src/ts/utils/dash3.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Utility functions to support both Dash 2 and Dash 3 components
+ *
+ * For more details, refer to the Dash documentation:
+ * Dash 3 for Component Developers - https://dash.plotly.com/dash-3-for-component-developers
+ */
+import React from "react";
+import { DashBaseProps } from "props/dash";
+
+/** check for dash version */
+export const isDash3 = (): boolean => {
+    return !!(window as any).dash_component_api;
+};
+
+/** Apply persistence settings based on React version */
+export const setPersistence = (Component: any, props: string[] = ["value"]): void => {
+    const persistence = { persisted_props: props, persistence_type: "local" };
+
+    if (parseFloat(React.version) < 18.3) {
+        Component.defaultProps = persistence;
+    } else {
+        Component.dashPersistence = persistence;
+    }
+};
+
+/** Get the loading state for the current component */
+export const getLoadingState = (loading_state?: DashBaseProps["loading_state"]): boolean => {
+    return isDash3()
+        ? (window as any).dash_component_api.useDashContext().useLoading()
+        : loading_state?.is_loading ?? false;
+};
+
+/** Get layout information for a child component */
+export const getChildLayout = (child: any): { type: any; props: any } => {
+    if (isDash3()) {
+        return (window as any).dash_component_api.getLayout(child.props.componentPath);
+    }
+
+    return {
+        type: child.props?._dashprivate_layout?.type,
+        props: child.props?._dashprivate_layout?.props,
+    };
+};
+
+/** Get only the props of a child component */
+export const getChildProps = (child: any): any => getChildLayout(child).props;
+
+/** Get the loading state of child components */
+export const getLoadingStateChildren = (
+    loading_state?: DashBaseProps["loading_state"],
+    children?: any
+): boolean => {
+    if (isDash3()) {
+        const ctx = (window as any).dash_component_api.useDashContext();
+        const childArray = React.Children.toArray(children);
+
+        // Get loading states for all children
+        const loadingStates = childArray.map((child: any) =>
```

Comment:
I think if you use `some` here rather than `map` you can benefit from short-circuit evaluation. I.e. you're return `true` the first time you encounter a loading child rather than calling `useLoading` on all children then calling `some` on the resultant array.

Reply:
Thanks!  Updated :slightly_smiling_face: